docs/source/dataframe-overview.rst

```diff
@@ -1,25 +1,42 @@
 Overview
 ========
 
-The ``dask.dataframe`` module implements a blocked parallel DataFrame that
-mimics a subset of the pandas DataFrame.  One dask DataFrame is comprised of
-several in-memory pandas DataFrames separated along the index.  An operation on
-one dask DataFrame triggers many pandas operations on the constituent pandas
-DataFrames in a way that is mindful of potential parallelism and memory
-constraints.
+Dask Dataframe implements a subset of the Pandas Dataframe interface using blocked
+algorithms, cutting up the large DataFrame into many small Pandas DataFrames.
+This lets us compute on dataframes that are larger than memory using all of our
+cores. We coordinate these blocked algorithms using dask graphs.
 
+Design
+------
+
+.. image:: images/dask-dataframe.svg
+   :alt: Dask DataFrames coordinate many Pandas DataFrames
+   :align: right
+   :width: 50%
+
+Dask dataframes coordinate many Pandas arrays arranged along an index.  These
+dataframes may live on disk or on other machines.
```

Comment:
> These dataframes may live on disk or on other machines.

Is this referring to the serialized dataframes on disk, or after being loaded as Dask dataframes, or...?